core/src/main/kotlin/org/jetbrains/research/testspark/core/test/kotlin/KotlinTestCompiler.kt

```diff
@@ -13,9 +13,12 @@ class KotlinTestCompiler(
     libPaths: List<String>,
     junitLibPaths: List<String>,
     kotlinSDKHomeDirectory: String,
+    javaHomeDirectoryPath: String,
 ) : TestCompiler(libPaths, junitLibPaths) {
     private val logger = KotlinLogging.logger { this::class.java }
     private val kotlinc: String
+    // Direct declaration in constructor requires changes to TestCompiler, so we declare it here.
+    private val javaHomeDirectoryPath = javaHomeDirectoryPath
```

Comment:
> requires changes to TestCompiler

What changes does it need? I found it visible to write like this and it worked fine:
```kotlin
class KotlinTestCompiler(
    ...,
    private val javaHomeDirectoryPath: String,
) : TestCompiler(..) {
  ..
}
```

Reply:
You are right, I have no clue what I was doing before. 😅 Thanks for pointing that out; it should be fixed now. 👍